### DISCUSSION TITLE
Submitted feature request:  Update entity generation to preserve customisations #1931

### DISCUSSION THREAD

**Discussion by dave42w** [SeaQL/sea-orm#1932] [Ideas]
Hi,

After discussions on Discord I have added a feature request that I am happy to code if the project is willing to accept it.

See [ Update entity generation to preserve customisations #1931 ](https://github.com/SeaQL/sea-orm/issues/1931)

I note that there have been multiple discussions on this topic in the past. For example:

1. [Preserve comments while generating entities with sea-orm-cli #1927](https://github.com/SeaQL/sea-orm/discussions/1927)
2. [use sea-orm-cli generate code will cover customed Enumeration #1897](https://github.com/SeaQL/sea-orm/discussions/1897)
3. [Skip Empty 'impl' Block Generation in 'sea-orm-cli generate entity' Command #1878](https://github.com/SeaQL/sea-orm/discussions/1878)

Thanks

Dave

**Comment by waynr** [comment]
It's possible to define impl blocks for types in other modules which seems to me like a pretty good approach for what I understand you to be trying to do. Eg

In a file named `src/entity.rs`:
```rust
#[derive(Default)]
pub(crate) struct Meow {
    pub(crate) count: u64
}
```

In a file named `src/entity_extensions.rs`:
```rust
impl super::entity::Meow {
    pub(crate) fn inc(&mut self) {
        self.count += 1;
    }

    pub(crate) fn get(&self) -> u64 {
        self.count
    }
}
```

in a file named `src/main.rs`:
```rust
mod entity;
mod entity_extensions;

use entity::Meow;

fn main() {
    let mut m = Meow::default();
    m.inc();
    m.inc();
    m.inc();
    println!("Hello, {}!", m.get());
}
```

For me this code produces the expected output:
```
Hello, 3!
```

With this approach you can put your entity extensions anywhere else in your crate.

**Reply by waynr** [reply]
Worth mentioning that this approach has the added advantage of cleanly separating the VCS history of regenerated code from the VCS history of customizations.
